### PR TITLE
	Extract context object out so it can be used by other features.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/Core/FindReferences/AbstractFindReferencesService.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ internal abstract partial class AbstractFindReferencesService :
             return Tuple.Create(result, project.Solution);
         }
 
-        private static string GetDisplayName(ISymbol symbol)
+        public static string GetDisplayName(ISymbol symbol)
         {
             return symbol.IsConstructor() ? symbol.ContainingType.Name : symbol.Name;
```
Comment (Pilchie) [inline src/EditorFeatures/Core/FindReferences/AbstractFindReferencesService.cs]: What about destructors, operators, etc.  Should this be some form of `Symbol.ToDisplayString()` instead?

File: src/EditorFeatures/Core/FindUsages/FindUsagesHelpers.cs [hunk 2]
```diff
@@ -44,5 +48,85 @@ internal static class FindUsagesHelpers
 
             return (mapping.Symbol, mapping.Project);
         }
+
+        public static async Task<(ISymbol symbol, Project project, ImmutableArray<ISymbol> implementations, string message)?> FindImplementationsAsync(Document document, int position, CancellationToken cancellationToken)
```
Comment (Pilchie) [inline src/EditorFeatures/Core/FindUsages/FindUsagesHelpers.cs]: Not sure I love this use of Tuples.

File: src/EditorFeatures/Core/FindUsages/FindUsagesHelpers.cs [hunk 3]
```diff
@@ -44,5 +48,85 @@ internal static class FindUsagesHelpers
...
+            ISymbol symbol, Project project, CancellationToken cancellationToken)
+        {
+            var solution = project.Solution;
+            if (symbol.IsInterfaceType() || symbol.IsImplementableMember())
+            {
+                var implementations = await SymbolFinder.FindImplementationsAsync(
+                    symbol, solution, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                // It's important we use a HashSet here -- we may have cases in an inheritence hierarchy where more than one method
+                // in an overrides chain implements the same interface method, and we want to duplicate those. The easiest way to do it
```
Comment (Pilchie) [inline src/EditorFeatures/Core/FindUsages/FindUsagesHelpers.cs]: Should this be "de-duplicate"?

**Pull request merged**